### DISCUSSION TITLE
How to specify some meta attributes with spacy package CLI?

### DISCUSSION THREAD

**Discussion by kinghuang** [explosion/spaCy#7756] [Help: Coding & Implementations] [answered]
With the [spacy package](https://spacy.io/api/cli#package) CLI command, how do I specify meta attributes like `description`, `author`, or `install_requires` for the generated Python module? If I create a `meta.json` file and pass it in, the resulting meta is missing a bunch of attributes like `pipeline` and `components` from the trained model.

**Created `meta.json` file:**

```json
{
  "author": "Enverus",
  "requirements": [
    "en_enverus_oi_lib",
    "en_enverus_oi_ner"
  ]
}
```

**With `meta.json` passed in (`spacy package --meta meta.json …`):**

```json
{
  "lang":"en",
  "name":"enverus_oi_nlp",
  "version":"0.0.0",
  "description":"",
  "author":"Enverus",
  "email":"",
  "url":"",
  "license":"MIT",
  "requirements":[
    "en_enverus_oi_lib",
    "en_enverus_oi_ner"
  ],
  "spacy_version":">=3.0.5,<3.1.0",
  "vectors":{
    "width":300,
    "vectors":684830,
    "keys":684830,
    "name":"en_vectors"
  }
}
```

**Without `meta.json` passed in (some data truncated to reduce verbosity):**

```json
{
  "lang":"en",
  "name":"enverus_oi_nlp",
  "version":"3.0.0",
  "description":"",
  "author":"",
  "email":"",
  "url":"",
  "license":"",
  "spacy_version":">=3.0.5,<3.1.0",
  "spacy_git_version":"53a3b967a",
  "vectors":{
    "width":300,
    "vectors":684830,
    "keys":684830,
    "name":"en_vectors"
  },
  "labels":{
    "tok2vec":[],
    "tagger":["$", "''", ","],
    "parser":["ROOT", "acl", "acomp"],
    "attribute_ruler":[],
    "lemmatizer":[],
    "entity-ruler-1":["PERCENT", "MONEY", "IDENT"],
    "entity-ruler-2":["PLURAL", "LOC", "DOC"],
    "ner":["ACTIVITY", "CARDINAL", "CHARGE"],
    "entity-ruler-3":["ORG"],
    "validate-ents":["GPE", "LOC", "UNIT"]
  },
  "pipeline":["tok2vec", "insights", "doc-optimizer"],
  "components":["tok2vec",  "insights", "doc-optimizer"],
  "disabled":[],
  "performance":{
    "tag_acc":0.0,
    "dep_uas":0.0,
    "dep_las":0.0,
    },
    "sents_f":0.8472906404,
    "ents_f":0.4139034987,
    "ents_p":0.3479430988,
    "ents_r":0.5107223476
  }
}
```

**Comment by svlandeg** [comment, marked as answer]
As you've found, the command defaults to `input_dir / "meta.json"` if a `meta_path` is not explicitely provided. If it is, then the one in the `input_dir` is actually ignored. The boolean `create_meta` does not help for this use-case either (this uses an interactive prompt).

I think the easiest workaround would be to have a little script that reads in the original file from `input_dir / "meta.json"`, changes the values you want to customize, write that out again, and use the resulting file as argument for `spacy package`?

**Reply by kinghuang** [reply]
Hmm, what about the other metadata from the trained model like the pipeline components, labels, and performance stats? It seems I'd have to manually read in the model's `meta.json`, modify it, and then pass it to the `spacy package` command.

It would be preferable if the `spacy package` command had an option to merge additional meta in with the model's existing meta, not replace it altogether.

**Reply by svlandeg** [reply]
Yea, I see your point! I'll look into this, it feels like it shouldn't be too much work to add this feature.

**Reply by svlandeg** [reply]
Quick update: this should be fixed by https://github.com/explosion/spaCy/pull/8663 :-)
